CRM/MembershipExtras/Hook/PostProcess/MembershipPaymentPlan.php

```diff
@@ -0,0 +1,418 @@
+<?php
+require_once('api/class.api.php');
+
+/**
+ * Post processes Membership Forms
+ */
+abstract class CRM_MembershipExtras_Hook_PostProcess_MembershipPaymentPlan {
+  /**
+   * @var CRM_Member_Form_Membership
+   *   Form object submitted to create a new membership.
+   */
+  protected $form;
+
+  /**
+   * @var object
+   *   Object with the created membership's data.
+   */
+  protected $membership;
+
+  /**
+   * @var object
+   *   Object with the contribution associated to the membership with full
+   *   payment info.
+   */
+  protected $membershipContribution;
+
+  /**
+   * @var object
+   *   Recurring contribution created for the payment plan.
+   */
+  protected $recurringContribution;
+
+  /**
+   * @var array
+   *   Array mapping status names to corresponding ID's.
+   */
+  protected static $contributionStatusValueMap = [];
+
+  /**
+   * CRM_MembershipExtras_Hook_PostProcess_Membership constructor.
+   *
+   * @param \CRM_Member_Form $form
+   */
+  public function __construct(CRM_Member_Form $form) {
+    $this->form = $form;
+  }
+
+  /**
+   * This method should check if the given action is the proper operation so
+   * that the form is post-processed.
+   *
+   * @param int $action
+   *   Constant value for the action, as defined in CRM_Core_Action class
+   *
+   * @return mixed
+   */
+  protected abstract function isCorrectOperation($action);

```

Comment:
same here, I think isAllowedOperation is a more clear name

Reply:
See response to prior comment ;)